test/integration/TaskServerFactoryTest.php

```diff
@@ -23,7 +25,8 @@ public function test_can_create_server_stack()
             $this->createStub(CacheItemPoolInterface::class),
             $this->createStub(TaskHandlerFactory::class),
             [],
-            []
+            [],
+            $this->createStub(TransactionMarkerMiddleware::class)
```

Comment:
I guess ideally we'd assert this middleware is actually registered (and at the end), tho it's not going to be that easy to do it.... Maybe just have to use ObjectPropertyRipper to inspect the internals in this case. Wasn't a concern before because all the other deps are required.

In which case possibly also worth a test with two extra middlewares to make sure the variadic doesn't get lost along the way in future.